test/unittest/tests/wrappers/mock_session.cpp

```diff
@@ -28,11 +28,16 @@ mock_session::~mock_session()
 {
     WT_CONNECTION_IMPL *connection_impl = _mock_connection->get_wt_connection_impl();
     if (_session_impl->block_manager != nullptr)
-        __wt_free(nullptr, _session_impl->block_manager);
+        _session_impl->block_manager_cleanup(_session_impl);
```

Comment:
This segfaults as `block_manager_cleanup` isn't set. I haven't figured out exactly why yet. I am trying to fix the issues in this test myself as I don't know the exact steps to get it to working.